File: 프로그래머스/3/42627. 디스크 컨트롤러/디스크 컨트롤러.py
def solution(jobs):
    answer, time = 0, 0
    jobs.sort(key = lambda x : x[1])
    n = len(jobs)
    
    while len(jobs) != 0:
        for i in range(len(jobs)):
            if jobs[i][0] <= time:
                time += jobs[i][1]
                answer += time - jobs[i][0]
                jobs.pop(i)
                break
            if len(jobs) - 1 == i:
                time += 1
    
    return answer // n

File: 프로그래머스/3/42627. 디스크 컨트롤러/디스크 컨트롤러.java
import java.util.*;

class Solution {
    public int solution(int[][] jobs) {
        int answer = 0;
        
        Arrays.sort(jobs, Comparator.comparingInt(a -> a[0]));
        PriorityQueue<int[]> pq = new PriorityQueue<>(Comparator.comparingInt(a -> a[1]));
        
        int time = 0;
        int idx = 0;
        while(idx < jobs.length || !pq.isEmpty()){
            while (idx < jobs.length && jobs[idx][0] <= time){
                pq.add(jobs[idx++]);
            }
            
            if (pq.isEmpty()){
                time = jobs[idx][0];
                continue;
            }
            
            int[] job = pq.poll();
            time += job[1];
            answer += time - job[0];
        }
        
        answer /= jobs.length;
        return (int) answer;
    }
}